Automattic/wp-calypso
# Sites Dashboard: Extract `ItemPreviewPane` as `ItemView`

Related to https://github.com/Automattic/dotcom-forge/issues/10138

## Proposed Changes

This PR extracts the existing `ItemPreviewPane` component into a new component called `ItemView`. The goal of this PR is to restructure the dependencies prior to redesign. 

- Make ItemPreviewPane shared
- Rename ItemPreviewPane -> ItemView, anticipating use for more than just the site view
- [Remove ItemPreviewPaneFooter](https://github.com/Automattic/wp-calypso/pull/97732/commits/abaaca4ce0f4bdc2867e08166599761e424d4f60) since it was not actually used. 

CC: @Automattic/a4a-genesis 

## Why are these changes being made?


- https://github.com/Automattic/dotcom-forge/issues/10138
- Please refer to the Project Thread pbxlJb-6J5-p2 for more context. 
- I made this component shared since I realize this component will continue to be used/shared for the site detail page after the redesign (green area below), while the “pane”-view will likely be retired and replaced with the full-page view.

| Dotcom | A4A | Dotcom after redesign |
|--------|--------|--------|
| <img width="1437" alt="Screenshot 2024-12-24 at 11 19 50" src="https://github.com/user-attachments/assets/ddc0cdce-8a69-4529-89fe-6241b28dda22" /> | <img width="1431" alt="Screenshot 2024-12-24 at 11 20 03" src="https://github.com/user-attachments/assets/ea6b95ed-a22c-47d4-bf5d-f8ce5a5d553f" /> | <img width="786" alt="Screenshot 2024-12-24 at 11 20 43" src="https://github.com/user-attachments/assets/2620bbf8-fa0a-4327-ae0d-bb0bbc57d2b6" /> | 

- We may make some changes in interfaces during the redesign. See the Slack convo p1734926657467739/1734659954.064219-slack-CRWCHQGUB.

## Testing Instructions



On both Calypso Live and A4A Live;

* Go to /sites
* Click any site
* Ensure nothing breaks

## Pre-merge Checklist
